tribe-ext-pdf-tickets/index.php

```diff
@@ -0,0 +1,876 @@
+	 *
+	 * @see Tribe__Extension__PDF_Tickets::get_pdf_url()
+	 *
+	 * @param $attendee_id
+	 *
+	 * @return string
+	 */
+	protected function ticket_link( $attendee_id ) {
+		$text = __( 'PDF Ticket', 'tribe-extension' );
+
+		$target = apply_filters( 'tribe_ext_pdf_tickets_link_target', '_blank' );
+
+		$output = sprintf(
+			'<a href="%s"',
+			esc_attr( $this->get_pdf_url( $attendee_id ) )
+		);
+
+		if ( ! empty( $target ) ) {
+			$output .= sprintf( ' target="%s"',
+				esc_attr( $target )
+			);
+		}
+
+		$output .= sprintf(
+			' class="tribe-ext-pdf-ticket-link">%s</a>',
+			esc_html( $text )
+		);
+
+		return $output;
+	}
+
+	/**
+	 * Add link to the PDF ticket to the front-end "View your Tickets" page.
+	 *
+	 * @see Tribe__Extension__PDF_Tickets::ticket_link()
+	 */
+	public function pdf_attendee_table_row_action_contents( $attendee ) {
+		echo $this->ticket_link( $attendee['attendee_id'] );
+	}
+
+
+	/**
+	 * Add a link to each ticket's PDF ticket on the wp-admin Attendee List.
+	 *
+	 * Community Events Tickets' Attendee List/Table comes from the same
+	 * source as the wp-admin one so no extra work to get it working there.
+	 *
+	 * @see Tribe__Extension__PDF_Tickets::ticket_link()
+	 */
+	public function pdf_attendee_table_row_actions( $actions, $item ) {
+		$actions[] = $this->ticket_link( $item['attendee_id'] );
+
+		return $actions;
+	}
+
+	/**
+	 * Outputs PDF.
+	 *
+	 * @see  Tribe__Extension__PDF_Tickets::get_mpdf()
+	 * @see  mPDF::Output()
+	 *
+	 * @link https://mpdf.github.io/reference/mpdf-functions/output.html
+	 *
+	 * @param string $html      HTML content to be turned into PDF.
+	 * @param string $file_name Full file name, including path on server.
+	 *                          The name of the file. If not specified, the
+	 *                          document will be sent to the browser
+	 *                          (destination I).
+	 *                          BLANK or omitted: "doc.pdf"
+	 * @param string $dest      I: send the file inline to the browser. The
+	 *                          plug-in is used if available.
+	 *                          The name given by $filename is used when one
+	 *                          selects the "Save as" option on the link
+	 *                          generating the PDF.
+	 *                          D: send to the browser and force a file
+	 *                          download with the name given by $filename.
+	 *                          F: save to a local file with the name given by
+	 *                          $filename (may include a path).
+	 *                          S: return the document as a string.
+	 *                          $filename is ignored.
+	 */
+	protected function output_pdf( $html, $file_name, $dest = 'F' ) {
+		// Should not happen but a fail-safe
+		if ( empty( $file_name ) ) {
+			$file_name = 'et_ticket_' . uniqid();
+		}
+
+		// If $file_name does not end in ".pdf", add it.
+		if ( '.pdf' !== substr( $file_name, - 4 ) ) {
```

Comment:
This shouldn't be necessary. I'd strip this out so that people can still output pdfs without the extensions when needed.

Reply:
done